src-tauri/src/utils/utils.rs

```diff
@@ -88,3 +90,35 @@ fn extract_port(config: &Value, key: &str) -> Result<u32> {
         .and_then(|v| v.parse().ok())
         .ok_or_else(|| AppError::Custom(format!("Failed to extract {} port", key)))
 }
+
+pub fn is_port_in_use(port: u32) -> bool {
+    match format!("127.0.0.1:{}", port).parse::<SocketAddr>() {
+        Ok(addr) => TcpStream::connect_timeout(&addr, Duration::from_secs(1)).is_ok(),
+        Err(_) => false,
+    }
+}
+
+pub fn check_ports_availability(config: &NodeConfig) -> Result<()> {
+    if is_port_in_use(config.server_port) || is_port_in_use(config.swarm_port) {
+        return Err(AppError::Custom(format!(
+            "Port {} or {} is already in use",
+            config.server_port, config.swarm_port
+        )));
+    }
+    Ok(())
+}
+
+pub fn kill_node_process(node_name: &str) -> std::io::Result<()> {
+    let output = Command::new("pkill")
+        .args(&["-f", &format!("meroctl.*--node-name {}.*run", node_name)])
+        .output()?;
```

Comment:
[`Command::spawn`](https://doc.rust-lang.org/std/process/struct.Command.html#method.spawn) returns a `Child` which has [`Child::kill`](https://doc.rust-lang.org/std/process/struct.Child.html#method.kill)

Reply:
The pkill fallback ensures process termination even if Child::kill fails or the process is not directly managed by the application.